shotover-proxy/src/transforms/util/cluster_connection_pool.rs

```diff
@@ -177,7 +177,10 @@ impl<C: Codec + 'static, A: Authenticator<T>, T: Token> ConnectionPool<C, A, T>
             .map_err(ConnectionError::IO)?;
 
         let mut connection = if let Some(tls) = &self.tls {
-            let tls_stream = tls.connect(stream).await.map_err(ConnectionError::TLS)?;
+            let tls_stream = tls
+                .connect_unverified_hostname(stream)
```

Comment:
Its concerning that we are making this call in a supposedly generic `util`.
Should we raise an issue to make this cluster logic redis specific?

Reply:
I don't think so, the method is pretty explicit that it does not verify the hostname. I'd say it's as clear as the `verify_hostname` method itself on `openssl::ssl::ConnectConfiguration`.

Reply:
right but you would have to check inside the cluster_connection_pool implementation to know that.
We can just leave at as is though, I dont mind.

Reply:
Oh true, I misinterpreted. Thought you were saying the `TlsConnector` was a generic util.

Reply:
When we look at Cassandra cluster support we can revisit that. We will find out whether we can have a generic cluster connection pool or if this needs to be a Redis specific one.